workflow/scripts/preprocess/dada2_trim.R

```diff
@@ -8,8 +8,7 @@ sink(loge, type = "message")
 
 library(dada2)
 
-in_R1 <- snakemake@input[["R1"]]
-in_R2 <- snakemake@input[["R2"]]
+in_R1 <- snakemake@input[["reads"]]
 
 out_R1 <- snakemake@output[["R1"]]
 out_R2 <- snakemake@output[["R2"]]
```

Comment:
So I take it that this doesn't behave strangely in the case of no "R2"?

Reply:
its a consequence of a limitation of snakemake input functions: afaik they can't returned named lists.  I'll try to clarify and remove misleading references to the names

Reply:
@nickp60 have you tried dicts and the unpack function? https://snakemake.readthedocs.io/en/v3.10.2/snakefiles/rules.html#input-functions-and-unpack

Reply:
@funnell I didn't know unpack preserved the names! Just tried it and it works; thank you!